docs/design/SYSTEM_DESIGN.md

```diff
@@ -6,101 +6,137 @@
-
-#### Bootstrap Config
-This is the configuration provided by the Infrastructure Administrator that allows them to bootstrap and configure various internal aspects of Envoy Gateway.
-It can be defined using a CLI argument similar to what [Envoy Proxy has](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-c).
-For e.g. users wanting to run Envoy Gateway in Kubernetes and use a custom [Envoy Proxy bootstrap config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-msg-config-bootstrap-v3-bootstrap) could define their Bootstrap Config as -
-```
-platform: kubernetes
-envoyProxy:
-  bootstrap: 
-    ......
-```
-
-#### User Config
-This configuration is based on the [Gateway API](https://gateway-api.sigs.k8s.io) and will provide:
-* Infrastructure Management capabilities to provision the infrastructure required to run the data plane, Envoy Proxy.
-This is expressed using [GatewayClass](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gatewayclass) and [Gateway](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) resources.
-* Ingress and API Gateway capabilities for the application developer to define networking and security intent for their incoming traffic.
-This is expressed using [HTTPRoute](https://gateway-api.sigs.k8s.io/concepts/api-overview/#httproute) and [TLSRoute](https://gateway-api.sigs.k8s.io/concepts/api-overview/#tlsroute).
-
-#### Workflow
-1. The Infrastructure Administrator spawns an Envoy Gateway process using a [Bootstrap Config](#bootstrap-config) to manage a fleet of Envoy Proxies.
-2. They will configure a [GatewayClass resource](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gatewayclass), that represents a class of Envoy Proxies.
-Envoy Gateway consumes this configuration and provisions a unique fleet of Envoy Proxies. the [GatawayClass parameters](https://gateway-api.sigs.k8s.io/v1alpha2/api-types/gatewayclass/#gatewayclass-parameters) section allows the infrastructure administrator to further modify attributes of the data plane. 
-3. They will configure a [Gateway resource](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) linking it to a specific GatewayClass 
-with information such as hostnames, protocol and ports stating which traffic flows are of interest.
-4. Application developers can now expose their APIs by configuring [HTTPRoute resources](https://gateway-api.sigs.k8s.io/concepts/api-overview/#httproute).
-These routes are attached to a specific [Gateway resource](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) allowing application traffic to reach
-the upstream application.
+Envoy Gateway is configured statically at runtime and the managed data plane is configured dynamically through
+Kubernetes resources, primarily [Gateway API][gw_api] objects.
+
+#### Static Configuration
+Static configuration is used to configure various internal aspects of Envoy Gateway at runtime, i.e. change the
```

Comment:
Would it be accurate to say this configures Envoy Gateway only at _startup time?_

Reply:
Yes, a user needs to restart EG for the runtime config to take effect. I see runtime/startup as two nouns with the same meaning.

Reply:
I think "startup" is definitely more specific here – configuration "at runtime" could mean any point while running, while "startup" makes it clear that you'll need a restart.

Reply:
+1 to startup because more specific.